x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -416,6 +417,40 @@ private LogicalPlan resolveEval(Eval eval, List<Attribute> childOutput) {
             return changed ? new Eval(eval.source(), eval.child(), newFields) : eval;
         }
 
+        /**
+         * resolve each item manually.
+         *
+         * Fields are added in the order they appear.
+         *
+         * If one field matches multiple expressions, the following precedence rules apply (higher to lower):
+         * 1. complete field name (ie. no wildcards)
+         * 2. partial wildcard expressions (eg. fieldNam*)
+         * 3. wildcard only (ie. *)
```

Comment:
The definition for `*` is "the rest of columns" or "all columns that do not match the other arguments". This description somewhat alludes to a different role for `*`.

Reply:
In a KEEP with multiple patterns, I think the natural meaning for `*` is "all the rest". It's a very specific and general meaning, that's why we gave it its own category with a specific (lower) priority, after the off-line conversations with @costin